### PR TITLE
#8 latest builds

### LINKED ISSUE #8: Latest builds Portlet
Das Standard-Portlet "Latest builds" soll getestet werden.

### PR DESCRIPTION
**Pull request by darenegade**
closes #8

### CODE REVIEW

**Comment by darenegade** [comment]
@peter-mueller @maximilian-zollbrecht 
Please review

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/LatestBuildsPortlet.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+/**
+ * The basic latest builds portlet shipped with the dashboard view plugin.
+ *
+ * @author Rene Zarwel
+ */
+@Describable("Latest builds")
+public class LatestBuildsPortlet extends AbstractDashboardViewPortlet {
+
+    private Control numberOfBuilds = control("numBuilds");
```
Comment (peter-mueller) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/LatestBuildsPortlet.java]: could be final?

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/LatestBuildsPortlet.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    private WebElement getRow(int row) {
+        return getTable().findElement(By.xpath(".//tbody/tr[" + row + "]"));
+    }
+
+    /**
+     * Sets the number of builds of the portlet.
+     *
+     * @param number the number of builds of the portlet
+     */
+    public void setNumberOfBuilds(int number) {
```
Comment (peter-mueller) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/LatestBuildsPortlet.java:59]: could throw illegal argument on negative values

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/LatestBuildsPortlet.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+     *
+     * @return latest builds table
+     */
+    public WebElement getTable() {
+        WebElement portlet = find(By.xpath("//div[contains(.,'" + PORTLET_NAME + "')]/following::table[1]"));
+
+        return portlet.findElement(By.id("statistics"));
+    }
+
+    private WebElement getRow(int row) {
```
Comment (peter-mueller) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/LatestBuildsPortlet.java:50]: could throw on bad parameter

**Review by peter-mueller — APPROVED** [review]
did my best to find something

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/LatestBuildsPortlet.java (hunk 3 above)
Comment (darenegade) [reply src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/LatestBuildsPortlet.java:50]: NoSuchElementException gets thrown already.....seems appropriate enough

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/LatestBuildsPortlet.java (hunk 2 above)
Comment (darenegade) [reply src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/LatestBuildsPortlet.java:59]: One can manually input bad values in the view as well, so i wouldn't restrict that. Maybe a test will especially check the behaviour on bad values in the future

**Pull request merged**